ngraph/core/reference/src/runtime/reference/concat.cpp

```diff
@@ -24,6 +24,20 @@ namespace ngraph
     {
         namespace reference
         {
+            namespace
+            {
+                std::vector<size_t> calculate_shape_sizes(const std::vector<Shape>& in_shapes)
+                {
+                    std::vector<size_t> sizes;
+                    sizes.reserve(in_shapes.size());
+                    for (size_t in_index = 0; in_index != in_shapes.size(); ++in_index)
+                    {
+                        sizes.push_back(shape_size(in_shapes[in_index]));
+                    }
+                    return sizes;
+                }
```

Comment:
Suggestion to use transform with back_inserter.
```suggestion
                std::vector<size_t> calculate_shape_sizes(const std::vector<Shape>& in_shapes)
                {
                    std::vector<size_t> sizes;
                    std::transform(in_shapes.begin(), in_shapes.end(), std::back_inserter(sizes), [](const Shape & shape) -> size_t { shape_size(shape) });
                    return sizes;
                }
```